clippy_lints/src/write.rs

```diff
@@ -234,20 +235,22 @@ fn check_tts<'a>(cx: &EarlyContext<'a>, tts: &ThinTokenStream, is_write: bool) -
         false,
         false,
     );
-    if is_write {
-        // skip the initial write target
-        parser.parse_expr().map_err(|mut err| err.cancel()).ok()?;
-        // might be `writeln!(foo)`
-        parser.expect(&token::Comma).map_err(|mut err| err.cancel()).ok()?;
-    }
+    // skip the initial write target
+    let expr: Option<Expr> = match parser.parse_expr().map_err(|mut err| err.cancel()).ok() {
+        Some(p) => Some(p.into_vec().0),
```

Comment:
`p` is just a [pointer](https://doc.rust-lang.org/nightly/nightly-rustc/syntax/ptr/struct.P.html)/a Box in disguise. You shouldn't need the `into_vec()` function. Please resolve the rest of the review first. If you still get this error, I will try to help you out more here. :wink: